src/lib/data/bluetooth.ts

```diff
@@ -57,43 +75,69 @@ export async function onPowerSensorEvent(
   });
 }
 
-async function connectToSensor(
-  sensorModel: SensorModel,
-  sensor: any,
-  sensorType: string,
-  callback: {
-    (r: RealtimeDataModel): Promise<RealtimeDataModel>;
-    (r: RealtimeDataModel): Promise<RealtimeDataModel>;
-  },
-) {
-  if (sensorModel === undefined) {
-    throw new Error(`No ${sensorType} sensor found`);
-  }
+// TODO type react-native-cycling-sensors
+export function sensorStatusService(sensor: any, sensorType: SensorType) {
+  // This will watch a sensor and watch a sensor for x interval
+  // to check it's connection and write that back to statusField.
+  let timer: NodeJS.Timer | null;
+  const statusField = getSensorStatusFieldFromType(sensorType);
+
+  const start = async (interval: number) => {
+    const record = await getOrCreateRealtimeRecord();
+
+    timer = setInterval(async () => {
+      // TODO I guess we should check in the db
+      // for any connected sensors and only show the
+      console.log(`checking ${sensorType} sensor`);
+      try {
+        // TODO we should upate react-native-cycling-sensors
+        // const isConnected = await sensor.isPeripheralConnected()
```

Comment:
Added a `.isConnected()` method on the sensors now which returns a promise<boolean>, its available in v0.6.0 of the lib: https://github.com/shaunmulligan/react-native-cycling-sensors/pull/21

Reply:
Sweet!